### DISCUSSION TITLE
why do I get 'PrismaClientUnknownRequestError'?

### DISCUSSION THREAD

**Discussion by Knorway** [prisma/prisma#12300] [Q&A] [answered]
I have an array of data received from client and just want to reverse their 'done' status by id

`data from client`
```
[
  { id: 1, text: 'prisma server', done: true },
  { id: 2, text: 'react query', done: true },
  { id: 3, text: 'work tomorrow', done: false }
]
```

`update logic`
```
router.post('/todos', async (req, res) => {
	try {
		const newTodos = req.body as Todo[];

		const updates = newTodos.map((todo) => {
			return prisma.todo.update({
				data: {
					text: todo.text,
					done: todo.done,
				},
				where: {
					id: todo.id,
				},
			});
		});

		await Promise.all(updates);

		res.status(201).end();
	} catch (error) {
		console.log(error);
	}
});
```

this is the `error` I get. I reckon this could be occurred from race condition but I don't think I'm creating multiple prisma instances and failed to find another way to update multiple item at once by id without using map and Promise.all.
```
PrismaClientUnknownRequestError: 
Invalid `prisma.todo.update()` invocation in
/Users/blake/Desktop/rq-startups/attempt2/server/app.ts:21:23

  18 try {
  19    const newTodos = req.body as Todo[];
  20    const updates = newTodos.map((todo) => {
→ 21            return prisma.todo.update(
  Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: ConnectionError(Timed out during query execution.) })
    at Object.request (/Users/blake/Desktop/rq-startups/attempt2/server/node_modules/@prisma/client/runtime/index.js:39813:15)
    at async PrismaClient._request (/Users/blake/Desktop/rq-startups/attempt2/server/node_modules/@prisma/client/runtime/index.js:40637:18)
    at async Promise.all (index 2) {
  clientVersion: '3.10.0'
}
```

any comments would be much helpful! thank you.

edited: I solved this using for of loop. the question is, is this the best way to use prisma in this case?
```
		for (const todo of newTodos) {
			await prisma.todo.update({
				data: {
					text: todo.text,
					done: todo.done,
				},
				where: {
					id: todo.id,
				},
			});
		}
```

**Comment by janpio** [comment, marked as answer]
How many entries are you trying to update at the same time?
That you are getting a pretty raw `PrismaClientUnknownRequestError` error is definitely a bug - can you please open a bug report issue? Thanks.

**Reply by Knorway** [reply]
I was trying to update only those 3 items and it also happens with 2 items. I'll post it on issue with more details. thank you for you answer!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
